perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ExpirationNotifScheduler.java

```diff
@@ -321,6 +332,20 @@ public void checkMembersState() {
 			log.error("Synchronizer: checkMembersState, attribute name is from wrong namespace.", e);
 		}
 
+		// check applications expiration in vos
+		try {
+			voApplicationsAutoRejection(vos);
```

Comment:
Probably not as a part of this pull-request, but we should really implement specific methods to get only such VOs and Groups, where there are expiration rules defined and which have some unprocessed applications.

Like this we are checking too much vos/group (like on BBMRI there is cca 6000 groups, but only a few have applications).

Also I feel like we should split this. We can have own class for application expiration and trigger it separately from the member expiration logic.

Reply:
I agree with both. But I am not sure we can now get attributes for entites where the attribute is set(at least I dont know about such metod?).

Also, there is a lot of things happenening in this class and it would be definitely worth splitting it up.

Reply:
There isn't such generic method, but if we split this logic, we can simply add new specific query method to get us just what we want - like when we get only members with expiration relative to passed date in the membership expiration logic.